web-api/src/storage/memory.rs

```diff
@@ -283,6 +283,11 @@ impl storage::Document for Storage {
         Ok(documents)
     }
 
+    async fn get_embedding(&self, id: &DocumentId) -> Result<Option<NormalizedEmbedding>, Error> {
+        let documents = self.documents.read().await;
+        Ok(documents.1.borrow_map().get(id).cloned())
```

Comment:
:art: there is no need for an intermediate assignment
```suggestion
        Ok(self.documents.read().await.1.borrow_map().get(id).cloned())
```

Reply:
readability, but I can change it.